x-pack/legacy/plugins/apm/public/components/app/Settings/CustomizeUI/CustomActionsOverview/CustomActionsFlyout/index.tsx

```diff
@@ -19,91 +14,133 @@ import {
+  onSave: () => void;
+  onDelete: () => void;
 }
 
-export const CustomActionsFlyout = ({ onClose }: Props) => {
-  const [serviceName, setServiceName] = useState('');
-  const [environment, setEnvironment] = useState('');
-  const [label, setLabel] = useState('');
-  const [url, setURL] = useState('');
-  return (
-    <EuiPortal>
-      <EuiFlyout ownFocus onClose={onClose} size="s">
-        <EuiFlyoutHeader hasBorder>
-          <EuiTitle size="s">
-            <h2>
-              {i18n.translate(
-                'xpack.apm.settings.customizeUI.customActions.flyout.title',
-                {
-                  defaultMessage: 'Create custom action'
-                }
-              )}
-            </h2>
-          </EuiTitle>
-        </EuiFlyoutHeader>
-        <EuiFlyoutBody>
-          <EuiText>
-            <p>
-              {i18n.translate(
-                'xpack.apm.settings.customizeUI.customActions.flyout.label',
-                {
-                  defaultMessage:
-                    "This action will be shown in the 'Actions' context menu for the trace and error detail components. You can specify any number of links, but only the first three will be shown, in alphabetical order."
-                }
-              )}
-            </p>
-          </EuiText>
-          <EuiSpacer size="l" />
-          <ServiceForm
-            isReadOnly={false}
-            serviceName={serviceName}
-            onServiceNameChange={setServiceName}
-            environment={environment}
-            onEnvironmentChange={setEnvironment}
-          />
+export interface CustomActionFormData extends Omit<CustomAction, 'filters'> {
+  filters: Filter[];
+}
 
-          <EuiSpacer size="l" />
+const convertFiltersToArray = (filters?: CustomAction['filters']) => {
+  if (filters && Object.keys(filters).length) {
+    return Object.keys(filters).map(key => {
+      return { key, value: filters[key] || '' };
+    });
+  }
+};
 
-          <SettingsSection
-            label={label}
-            onLabelChange={setLabel}
-            url={url}
-            onURLChange={setURL}
-          />
-        </EuiFlyoutBody>
-        <EuiFlyoutFooter>
-          <EuiFlexGroup justifyContent="spaceBetween">
-            <EuiFlexItem grow={false}>
-              <EuiButtonEmpty iconType="cross" onClick={onClose} flush="left">
+const convertFiltersToObject = (
+  filters: CustomActionFormData['filters']
+): CustomAction['filters'] => {
+  if (filters && filters.length) {
+    return filters
+      .filter(({ key, value }) => !isEmpty(key) && !isEmpty(value))
+      .reduce((acc: Record<string, string>, { key, value }) => {
+        acc[key] = value;
+        return acc;
+      }, {}) as CustomAction['filters'];
+  }
+};
+
+export const CustomActionsFlyout = ({
+  onClose,
+  customActionSelected,
+  onSave,
+  onDelete
+}: Props) => {
+  const callApmApiFromHook = useCallApmApi();
+  const [isLoading, setIsLoading] = useState(false);
```

Comment:
To be consistent with the `isDeleting` somewhere else:
```suggestion
  const [isSaving, setIsSaving] = useState(false);
```